### PR TITLE
Add metrics sever

### PR DESCRIPTION
**Pull request by ddellarocca**
## What this PR is for?
Addition of the metrics-server.
This PR adds the metrics-server module.

### CODE REVIEW

File: modules/observability/metrics-server/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Changelog
+
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
+and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## Unreleased
+
+## v1.26.0
```
Comment (JGiola) [inline modules/observability/metrics-server/CHANGELOG.md]:
```suggestion
```

We don't aim to release the module for the 1.26 version, so we can remove this from the changelog

File: modules/observability/metrics-server/README.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+This module use the following user, gid and fsGroup:
+
+- metrics-server: **48090**
+
+## Flavors
+
+### Secure kubelet
+
+- check TLS when to nodes
```
Comment (JGiola) [inline modules/observability/metrics-server/README.md]:
```suggestion
- check TLS validity when connecting to the kubelet
```

File: modules/observability/metrics-server/base/resources/workloads/metrics-server/api-service.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+apiVersion: apiregistration.k8s.io/v1
+kind: APIService
+metadata:
+  name: v1beta1.metrics.k8s.io
+spec:
+  group: metrics.k8s.io
+  groupPriorityMinimum: 100
+  insecureSkipTLSVerify: true
```
Comment (JGiola) [inline modules/observability/metrics-server/base/resources/workloads/metrics-server/api-service.yaml]: There is a way to allow us to flip this flag to true?

File: modules/observability/metrics-server/base/resources/rbac/clusterroles.yaml [hunk 4]
```diff
@@ -0,0 +1,39 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  labels:
+    rbac.authorization.k8s.io/aggregate-to-admin: "true"
+    rbac.authorization.k8s.io/aggregate-to-edit: "true"
+    rbac.authorization.k8s.io/aggregate-to-view: "true"
```
Comment (JGiola) [inline modules/observability/metrics-server/base/resources/rbac/clusterroles.yaml]: Minor thing, based on how the aggregation works we can leave only the `aggregate-to-view` label and the others will pick it up the same
```suggestion
    rbac.authorization.k8s.io/aggregate-to-view: "true"
```

File: modules/observability/metrics-server/base/resources/workloads/kustomization.yaml [hunk 5]
```diff
@@ -0,0 +1,4 @@
+kind: Kustomization
+apiVersion: kustomize.config.k8s.io/v1beta1
```
Comment (JGiola) [inline modules/observability/metrics-server/base/resources/workloads/kustomization.yaml:2]:
```suggestion
apiVersion: kustomize.config.k8s.io/v1beta1
commonLabels:
  app.kubernetes.io/instance: metric-server
  app.kubernetes.io/name: metric-server
```

Can we add this two additional labels to uniform it to the other modules?

File: modules/observability/metrics-server/base/resources/workloads/metrics-server/api-service.yaml
Comment (JGiola) [inline modules/observability/metrics-server/base/resources/workloads/metrics-server/api-service.yaml:1]: I was thinking if is not better to handle this file like a crd, and move it at the top of the modules inside an `apiservices` folder...

**Review by JGiola — CHANGES_REQUESTED** [review]
Thank you for the module, I've added some minor comments and suggestions before merging it.

File: modules/observability/metrics-server/base/resources/workloads/metrics-server/api-service.yaml (hunk 3 above)
Comment (ddellarocca) [reply modules/observability/metrics-server/base/resources/workloads/metrics-server/api-service.yaml]: Tried but it is not working. We can try to use cert-manager in order to generate and mange its certificate.

Comment (JGiola) [reply modules/observability/metrics-server/base/resources/workloads/metrics-server/api-service.yaml]: We can left if for a future improvement for now

**Review by JGiola — APPROVED** [review]

**Pull request merged**